Troubleshoot/ihp-troubleshoot

```diff
@@ -1,120 +1,229 @@
+        "ghci is not loaded from /nix/store/... Is direnv hooked into your shell?"
+    )
+
+show_step("Checking .ghci")
+
+path_exists(
+    ".ghci",
+    "Found .ghci",
+    ".ghci missing"
+)
+
+ghci_output, _ = run_command(["ghci", "2>&1"], "putStrLn \"ok\"")
+
+if ".ghci is writable by someone else" in ghci_output:
+    show_failure(
+        ".ghci permissions are wrong. Try 'chmod go-w .ghci' to fix this"
+    )
+else:
+    show_success(".ghci permissions are ok")
+
+show_step("Checking IHP")
+
+ihp_lib_is_link = path.islink("build/ihp-lib")
+
+if ihp_lib_is_link:
+    show_success("Symlink build/ihp-lib exists")
+else:
+    show_failure(
+        "Symlink build/ihp-lib is missing. Try 'make build/ihp-lib' to fix this"
+    )
+
+ihp_lib_link = readlink("build/ihp-lib") if ihp_lib_is_link else None
+
+path_exists(
+    ihp_lib_link,
+    "Symlink build/ihp-lib target exists",
+    "Symlink build/ihp-lib target directory does not exist. Try `nix-shell --run \"make -B build/ihp-lib\"` to fix this"
+)
+
+if path.exists("IHP"):
+    show_success("Local IHP directory exists")
+else:
+    show_success("IHP used from nix")
+
+show_step("Checking Cachix")
+
+cachix_output, _ = run_command(["cachix", "--help"])
+
+if "https://cachix.org command line interface" in cachix_output:
+    show_success("Cachix exists")
+else:
+    "Cachix is missing. Is cachix installed?"
+
+with open(path.join(Path.home(), ".config/nix/nix.conf")) as file:
+    nix_conf = file.read()
+
+    if "digitallyinduced.cachix.org-1:y+wQvrnxQ+PdEsCt91rmvv39qRCYzEgGQaldK26hCKE=" in nix_conf:
+        show_success("digitallyinduced.cachix.org configured")
+    else:
+        show_failure(
+            "digitallyinduced.cachix.org binary cache missing. Try 'cachix use digitallyinduced'"
+        )
+
+    if "digitallyinduced.cachix.org-1:3mGU1b6u5obFp2VUfI55Xe8/+mawl7y9Eztu3rb94PI=" in nix_conf:
+        show_failure(
+            "Found legacy cachix public key for digitallyinduced.cachix.org. Try to remove digitallyinduced.cachix.org-1:3mGU1b6u5obFp2VUfI55Xe8/+mawl7y9Eztu3rb94PI= from ~/.config/nix/nix.conf"
+        )
+    else:
+        show_success("No legacy cachix key found")
+
+show_step("Checking IHP files")
+
+SENSITIVE_FILES = [".ghci", "start", "hie.yaml", "default.nix"]
+with request.urlopen("https://api.github.com/repos/digitallyinduced/ihp-boilerplate/contents/") as response:
+    if response.status is not 200:
+        show_failure("Failed to send GET request to GitHub API")
+    else:
+        deserialized = load(response)
+        for file_details in deserialized:
+            file_path = file_details["path"]
+            if file_path in SENSITIVE_FILES:
+                local_file_sha, _ = run_command(
+                    ["git", "hash-object", file_path])
+                if local_file_sha.rstrip() != file_details["sha"]:
+                    show_warning(
+                        f"Make sure the `{file_path}` file is same as {file_details['html_url']}"
```

Comment:
I'd change this message to something like `The file file_path was edited manually. Try reverting your changes to html_url to see whether that affects the issue`